once_test.py

```diff
@@ -360,6 +456,70 @@ def value():
         self.assertEqual(_CallOnceClass.value(), 1)
         self.assertEqual(_CallOnceClass.value(), 1)
 
+    def test_receiving_iterator(self):
+        @once.once
+        def receiving_iterator():
+            next = yield 0
+            while next is not None:
+                next = yield next
+
+        gen_1 = receiving_iterator()
+        gen_2 = receiving_iterator()
+        self.assertEqual(gen_1.send(None), 0)
+        self.assertEqual(gen_1.send(1), 1)
+        self.assertEqual(gen_1.send(2), 2)
+        self.assertEqual(gen_2.send(None), 0)
+        self.assertEqual(gen_2.send(-1), 1)
+        self.assertEqual(gen_2.send(-1), 2)
+        self.assertEqual(gen_2.send(5), 5)
+        self.assertEqual(next(gen_2, None), None)
+        self.assertEqual(gen_1.send(None), 5)
+        self.assertEqual(next(gen_1, None), None)
+        self.assertEqual(list(receiving_iterator()), [0, 1, 2, 5])
+
+    def test_receiving_iterator_parallel_execution(self):
```

Comment:
Oh I also think it would be good to have a sync analog to `test_iterator_lock_not_held_during_evaluation`

Reply:
will do in upcoming refactor